File: app/src/main/java/com/alderferstudios/percentcalculatorv2/PreferenceDoubles.java
package com.alderferstudios.percentcalculatorv2;

import android.content.SharedPreferences;

/**
 * Percent Calculator
 * Utility class for putting and getting doubles in SharedPreferences
 * <p>
 * Alderfer Studios
 *
 * @author Ben Alderfer
 */
public class PreferenceDoubles {

    /**
     * Converts a Double to a Long and puts it into the editor
     * @param edit the editor
     * @param key the key for future reference
     * @param value the double to put
     * @return the editor with the Double added
     */
    public static SharedPreferences.Editor putDouble(final SharedPreferences.Editor edit, final String key, final double value) {
        return edit.putLong(key, Double.doubleToRawLongBits(value));
    }

    /**
     * Used to get the Double from a saved Long
     * @param prefs the preference object
     * @param key the key for future reference
     * @param defaultValue the defaultValue if nothing is there
     * @return the double value
     */
    public static double getDouble(final SharedPreferences prefs, final String key, final double defaultValue) {
        if (!prefs.contains(key)) {
            return defaultValue;
        }

        return Double.longBitsToDouble(prefs.getLong(key, 0));
    }
}

File: app/src/main/java/com/alderferstudios/percentcalculatorv2/util/PreferenceDoubles.kt
package com.alderferstudios.percentcalculatorv2.util

import android.content.SharedPreferences

/**
 * Utility class for putting and getting doubles in SharedPreferences
 */
object PreferenceDoubles {

    /**
     * Converts a Double to a Long and puts it into the editor
     * @param edit the editor
     * @param key the key for future reference
     * @param value the double to put
     * @return the editor with the Double added
     */
    fun putDouble(edit: SharedPreferences.Editor, key: String, value: Double): SharedPreferences.Editor {
        return edit.putLong(key, java.lang.Double.doubleToRawLongBits(value))
    }

    /**
     * Used to get the Double from a saved Long
     * @param prefs the preference object
     * @param key the key for future reference
     * @param defaultValue the defaultValue if nothing is there
     * @return the double value
     */
    fun getDouble(prefs: SharedPreferences?, key: String?, defaultValue: Double): Double {
        return if (prefs == null || key == null || !prefs.contains(key)) {
            defaultValue
        } else java.lang.Double.longBitsToDouble(prefs.getLong(key, 0))

    }
}
